src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -179,7 +196,50 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
 
 		return new Timespan(t);
 	}
-	
+
+
+	/**
+	 * Parses a shortened timespan (e.g., "5m", "3d", "1h") based on the provided input string and parse context.
+	 * Returns null if the input does not represent a valid shortened timespan in the context of a command argument.
+	 *
+	 * @param input   The input string representing the shortened timespan.
+	 * @param context The parse context to determine if parsing is allowed.
+	 * @return A Timespan object representing the parsed timespan, or null if parsing fails or not allowed in context.
+	 */
+	@Nullable
+	public static Timespan parseShortenedTimespan(String input, ParseContext context) {
+		if (context != ParseContext.COMMAND) {
+			return null;
+		}
```

Comment:
No need for this context check. If a user is calling this function, they are doing so on purpose. Simply asking for a key that's never actually used seems silly.